### ISSUE TITLE
Sync data on cloud

### ISSUE THREAD

**Issue by Laffery** [yrming/vscode-type-challenges#17]
Awesome extension it is!

I have two or more devices, which are for work or for life. Unfortunately, the `workspaceFolder` is local, and I have to create GitHub repository to sync my challenges data from different devices.
Is there any much more convenient way to sync data on cloud? Or any plan to support this feature in your roadmap?

**Comment by yrming** [comment]
@Laffery Thank you very much for your suggestion, I am thinking about how to implement this feature perfectly.

**Comment by dabuside** [comment]
What about sync completed tasks to the cloud?
example sync data
```
{
   completedList: [1, 100, 2]
}
```

**Comment by meta-panic** [comment]
Do we want to sync only progress or the solutions as well?

https://github.com/shanalikhan/code-settings-sync - the extension(it is deprecated) used github oauth to store its information in secret gists
so it is possible to hold the solution there as well

@yrming what do you think?

**Comment by yrming** [comment]
> Do we want to sync only progress or the solutions as well?
> 
> [shanalikhan/code-settings-sync](https://github.com/shanalikhan/code-settings-sync) - the extension(it is deprecated) used github oauth to store its information in secret gists so it is possible to hold the solution there as well
> 
> @yrming what do you think?

@meta-panic @dabuside From my perspective, I think sync solutions might be more perfect. If we only sync progress, when users click on a completed challenge, they can't see their own solution to the challenge, which feels a bit weird.

**Comment by dabuside** [comment]
> > Do we want to sync only progress or the solutions as well?
> > [shanalikhan/code-settings-sync](https://github.com/shanalikhan/code-settings-sync) - the extension(it is deprecated) used github oauth to store its information in secret gists so it is possible to hold the solution there as well
> > @yrming what do you think?
> 
> @meta-panic @dabuside From my perspective, I think sync solutions might be more perfect. If we only sync progress, when users click on a completed challenge, they can't see their own solution to the challenge, which feels a bit weird.

I agree with you. Sync progress and all done or draft solutions will be helpful If vscode-setting-sync or github-gist allow us to upload those data to cloud.
